lib/galaxy/util/plugin_config.py

```diff
@@ -8,6 +9,9 @@
 from galaxy.util.submodules import import_submodules
 
 
+PluginConfigSource = collections.namedtuple('PluginConfigSource', ['type', 'source'])
```

Comment:
🎩

Reply:
I strongly dislike accessing non-local tuples by index.

Reply:
It was a good tophat 🤣

Reply:
Shouldn't also `load_plugins()` been modified at line 40 to use the keys?

Reply:
@nsoranzo There are other callers still passing a tuple apparently.

Reply:
Are there? I couldn't find any in `lib/galaxy/job_metrics/__init__.py` , `lib/galaxy/tool_util/deps/__init__.py` , `lib/galaxy/tool_util/deps/containers.py` or `lib/galaxy/tools/error_reports/__init__.py` .

Reply:
[Here's one](https://github.com/galaxyproject/galaxy/blob/5c1380fff60a73261d37ce3a9cb53e5ede56416c/lib/galaxy/tool_util/deps/__init__.py#L126).

Reply:
This is going to conflict https://github.com/galaxyproject/galaxy/pull/8616/files isn't it, I shouldn't have taken so long to respond to the review of it.

Reply:
https://github.com/galaxyproject/galaxy/pull/8616/commits/28b423b78b8050f9f26d4591a22365736f4ec61f